api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1780,16 +1789,14 @@ type AzurePlatformSpec struct {
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 
-	// SecurityGroupName is the name of an existing security group on SubnetName. This field is provided as part of the
+	// SecurityGroupID is the ID of an existing security group on the SubnetID. This field is provided as part of the
 	// configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM).
 	//
-	// Example: if your Network Security Group ID is /subscriptions/<subscriptionID>/resourcegroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<securityGroupName>,
-	//          your SecurityGroupName is <securityGroupName>
-	//
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupName is immutable"
-	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupID is immutable"
+	// +kubebuilder:validation:Required
+	// +required
 	// +immutable
-	SecurityGroupName string `json:"securityGroupName,omitempty"`
+	SecurityGroupID string `json:"securityGroupID,omitempty"`
```

Comment:
Here I would document the assumptions / requirements that are being made.
* The Subscription part specified in the SecurityGroupID has to match the value specified in the SubscriptionID attribute. If that's not the case let me know
* Does the resource group name part of the SecurityGroupID needs to match the ResourceGroupName value or the NetworkResourceGroupName value?

Reply:
Added documentation to address both points